### ISSUE TITLE
{{ . }} notation doesn't work with sets

### ISSUE THREAD

**Issue by dluksza** [davidsantiago/stencil#30]
The `{{ . }}` notation doesn't work for sets. Instead of iterate over data in set it will just call `str` function. Here is example of current behaviour:
```clojure
(render-string "{{# foo }} {{ . }} {{/ foo }}" {:foo #{1 2 3}})
```
will return:
```
" #{1 3 2} "
```
In this case I expect to get:
```
" 1  2  3 "
```

**Comment by davidsantiago** [comment]
Yeah, I'm not sure if this should be implemented. Though you expect "1 2 3", a set actually doesn't have a defined order. So while clojure will let you call seq on a set to get the elements as a sequence, you may get the elements in any order. And they may even be different orders across different runs of the program. Stencil currently requires a list for a section iterator to be a sequential (that is, calling sequential? on it returns true). This is a pretty minimal requirement for satisfying mustache's idea of a list for a section. If you have some object that isn't sequential but has an ordering that you'd be satisfied with, you can always call the conversion before sticking it into your context (in this case, {:foo (seq #{1 2 3})}. I generally think the occasional need for additional specificity from the user is worthwhile in order to get clear behavior.

**Comment by dluksza** [comment]
I don't expect to get particular order from iterating over set, but rather consistent behaviour, since it works for vectors I don't see any reason why it shouldn't also work for sets.

Of course one should be aware that order will not be persisted.

Currently I'm calling (into []) on set just before it is added into context map. It would be nice if this conversion will not be required.
